test-framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -161,6 +161,25 @@ public static void restoreContentType() {
         Requests.INDEX_CONTENT_TYPE = XContentType.JSON;
     }
 
+    private static boolean TESTS_SEED_PROPERTY_SET = false;
+    @BeforeClass
+    @SuppressForbidden(reason = "sets tests.seed")
```

Comment:
and setting it here is too late anyway, for example StringHelper in lucene is already initialized. changing tests.seed is simply a non-option. If you want things to reproduce in your IDE, you need to pass tests.seed to tests: its just that simple.